### ISSUE TITLE
[ECS] [request]: Add UDP_TCP protocol for container portMapping in the task definition

### ISSUE THREAD

**Issue by abhishek181** [aws/containers-roadmap#850]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Tell us about your request**
The port mapping option for the containers currently supports TCP or UDP protocol. It should also have an option for protocol UDP_TCP. This would allow the same dynamic port to be assigned to the same container port over TCP and UDP.

**Which service(s) is this request for?**
ECS

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
Currently, there is no option to map the same container port with different protocols. So, while trying to create a service behind an UDP Network Load Balancer the health checks fail as the health checks are done on TCP. If the UDP_TCP protocol is enabled then the same container port will be mapped to a single dynamic host port and would be helpful for application that serve traffic over both TCP and UDP.

**Are you currently working around this issue?**
Currently to solve this issue, a DAEMON service is also has to be deployed and the health check port of the NLB has to be changed to point to the daemon service port. The health check port has to be reverted back to "traffic-port" while updating the REPLICA service(behind the NLB) and then again the health check port is pointed to the daemon service port

**Comment by viesti** [comment]
This would be really great. Alternatively, if there would be a `"health-port"`, in addition `"traffic-port"`, which could be mapped to a container port, to allow running a health checker on a separate, dynamically allocated port. This might require support from the team working on the load balancer though.

**Comment by frankyi-gh** [comment]
Ran into this issue while trying to deploy a cluster of prometheus-alertmanagers in Fargate. Alertmanager uses [a gossip protocol](https://github.com/hashicorp/memberlist) that requires UDP and TCP on the same port. 

Unfortunately, for Fargate, this is a blocker with no workaround. Using something like `iptables` to reroute one protocol is not possible because you need the `NET_ADMIN` Docker capability, which is not available in Fargate.

**Comment by justinhauer** [comment]
Looks like over 100 customers are asking for this, and it's been open for nearly 3 years, how can this issue get priority on the roadmap?

**Comment by dcrawkstar** [comment]
even more waiting....

**Comment by justinhauer** [comment]
The wait is real.

**Comment by HeroesDieYoung** [comment]
Needs some noise to wake it up. I haven't worked on the project I needed this for for almost two years, but I'd still like to see the feature get delivered

**Comment by mrudel** [comment]
We are running into this with a Consul deployment as well, it would prefer to use UDP and TCP for gossip on the same port. Just a little extra noise.

**Comment by alexs20** [comment]
We are also waiting... already more than 4 years!

**Comment by danhiris** [comment]
To make matters worse, when trying to set up a nlb using IP target group, I am forced to set up a health check, and UDP isn't valid for health checks. So I need to have both UDP and TCP port open for the health check to work on TCP, but since I can't set the ECS listener on both UDP and TCP, I'm in a catch 22 where I can't run a UDP service in ECS.

**Comment by anitakrueger** [comment]
Coming up on 5 years and we have to create EC2 instances and CodeDeploy apps because of this limitation. All our other services are Fargate. Has anyone brought this up with AWS Support or their TAM?

**Comment by sepehrnour** [comment]
Still running into this issue half a decade since this ticket was opened.

**Comment by Trevato** [comment]
+1

**Comment by Becram** [comment]
Waiting for it

**Comment by rocky-foreflight** [comment]
Still waiting.

**Comment by rocky-foreflight** [comment]
@LukeCarrier what's your deal?

**Comment by alexs20** [comment]
That’s the purpose of it - to send notifications to repo owners and move
the thread to the top, letting developers know we’re still waiting. If you
don’t want to receive notifications, simply unsubscribe from the thread.

**Comment by LukeCarrier** [comment]
Spamming this thread isn't making the blindest bit of difference. It is very possible that this behaviour is actively putting off anyone from picking this up.

Contact your AWS account manager, or escalate with support.

**Comment by azN2** [comment]
If it doesn't increase spend. Its deprioritized

**Comment by devopstravelwifi** [comment]
Any update on this?

**Comment by vivekgupta-3838** [comment]
Any update on this?

**Comment by FarrOut** [comment]
yes please. i have stopped using ECS because of this very issue.
